src/imageService/index.js

```diff
@@ -0,0 +1,45 @@
+import ImagePicker from 'react-native-image-picker'
+import CameraError from '~/AppErrors/CameraError'
+
+const selectImageFromGalleryOrCamera = () => showImagePickerPromiseWrapper(options)
+
+const options = {
+  mediaType: 'photo',
+  title: 'Seleccionar una foto',
+  takePhotoButtonTitle: 'Tomar una foto',
+  chooseFromLibraryButtonTitle: 'Seleccionar desde galería',
+  cancelButtonTitle: 'cancelar',
+  noData: true,
+  storageOptions: {
+    skipBackup: true,
+    path: 'CannabisDiagnose',
+    cameraRoll: false
+  }
+}
+
+const showImagePickerPromiseWrapper = (options) => (
+  new Promise((resolve, reject) => {
+    ImagePicker.showImagePicker(options, (imageResponse) => {
+      if (!imageResponse.uri) {
+        reject(new CameraError('Image not taken nor selected.'))
+      }
+      resolve(buildImageRepresentation(imageResponse))
```

Comment:
Careful! You need to return in the `if` branch or add an `else`. Otherwise, you'll call both `reject` and `resolve` in case the `imageResponse` doesn't contain an `uri`.